src/config/authConfig.js

```diff
@@ -0,0 +1,36 @@
+export const getAuthQuery = (dialog = false) => {
+  const authEndpoint = 'https://accounts.spotify.com/authorize';
+  const clientId = '10a41ddfc787418f9ef272f0bf886e86';
+  const redirectUri = 'http://localhost:8080/';
+  const scopes = [
+    'playlist-read-private',
+    'playlist-modify-public',
+    'playlist-modify-private',
+  ];
+  const responseType = 'token';
+  const showDialog = dialog;
+
+  const authQuery = `${authEndpoint}?client_id=${clientId}&redirect_uri=${redirectUri}&scope=${scopes.join('%20')}&response_type=${responseType}&show_dialog=${showDialog}`;
+  return authQuery;
+};
+
+export const getTokenFromURI = () => {
+  const hash = window.location.hash
+    .substring(1)
+    .split('&')
+    .reduce((initial, item) => {
+      if (item) {
+        const parts = item.split('=');
+        initial[parts[0]] = decodeURIComponent(parts[1]);
+      }
+      return initial;
+    }, {});
+  window.location.hash = '';
+
+  return hash.access_token || null;

```

Comment:
Why `null` and not `false`?

Reply:
I've been using `null` because I understood it to be something you use as a placeholder for a value that is expected but just doesn't exist yet. My reasoning for using it here and in state was that since `App.state.token` is ultimately going to hold a string, I shouldn't initialize it or set it to `false` because it's not meant to be a boolean. Now reading up on it I'm not seeing much support for `null` so I'm not sure where I got that idea.